src/index.js

```diff
@@ -0,0 +1,367 @@
+                this.setState({playWithComputer: !this.state.playWithComputer})}
+              }
+            }
+          />
+          <FirstStepRight
+            lock={!this.state.stepNumber ? '' : 'lock'}
+            hide={this.state.playWithComputer ? '' : 'hide'}
+            manFirst={() => {
+              this.resetState();
+              this.setState({
+                playerGoesFirst: true,
+                xIsNext: true,
+              })
+            }}
+            computerFirst={() => {
+              this.resetState();
+
+              const squares = Array(9).fill(null);
+              const squareNumber = Logic.firstStepAlgorithm(squares, false);
+              squares[squareNumber] = 'X';
+              this.setState({
+                history: [{
+                  squares: squares,
+                }],
+                playerGoesFirst: false,
+                xIsNext: false,
+              });
+            }}
+          />
+          <button
+            className={`game-restart ${!this.state.stepNumber ? 'hide' : ''}`}
+            onClick={(e) => {
+                this.resetState();
+
+                const btn = document.querySelector('.game-mode');
+                btn.classList.add('active');
+
+                const firstStepBtns = document.querySelectorAll('.first-step button');
+                firstStepBtns.forEach(btn => btn.classList.remove('active'));
+              }
+            }
+          >Start new game
+          </button>
+          <button
+            className={`game-reverse ${!this.state.stepNumber ? 'hide' : ''}`}
+            onClick={() => this.setState({reverseListOfMoves: !this.state.reverseListOfMoves})}
+          >Reverse list
+          </button>
+          <ol className={`game-list ${!this.state.stepNumber ? 'hide' : ''}`}>
+            <li className="game-list-toggle">
+              <button
+                onClick={(e) => {
+                  const list = e.currentTarget.closest('ol');
+
+                  if (list.classList.contains('open')) {
+                    list.classList.remove('open');
+                  } else {
+                    list.classList.add('open');
+                  }
+                }}
+              >
+                List of Moves
+              </button>
+            </li>
+            {moves}
+          </ol>
+        </div>
+      </div>
+    );
+  }
+}
+
+class Board extends React.Component {
+  renderSquare(i, className) {
+    return (
+      <Square
+        className={className || 'square'}
+        value={this.props.squares[i]}
+        onClick={() => this.props.onClick(i)}
+        key={i}
+      />
+    )
+  }
+
+  render() {
+    let squares = [];
+    let rows = [];
+
+    for (let i = 0; i < 9; i++){
+      let className = 'square';
+
+      const winnerLine = this.props.winnerLine;
+      if (winnerLine && winnerLine.includes(i)) {
+        className = 'square square-win';
+      }
+
+      squares.push(this.renderSquare(i, className));
```

Comment:
adapt to not use methods like `push`, `pop`, `shift` and etc in React, all methods that mutate data should be avoided

Reply:
Can I fixed it like this? :
```
const squares = Array(9).fill(null).map((square, index) => {
  let className = 'square';

  const {winnerLine} = this.props;
  if (winnerLine?.includes(index)) {
    className = 'square square-win';
  }

  return this.renderSquare(index, className);
})
```